client/src/main/java/io/pravega/client/segment/impl/SegmentInputStreamImpl.java

```diff
@@ -121,6 +127,10 @@ public ByteBuffer read(long timeout) throws EndOfSegmentException, SegmentTrunca
     }
 
     private ByteBuffer readEventData(long timeout) throws EndOfSegmentException, SegmentTruncatedException {
+        if (this.offset == this.endOffset) {
+            log.debug("All events up to the configured end offset:{} have been read", endOffset);
+            return null;
```

Comment:
Returning null is not the most informative thing. Perhaps it should throw EndOfSegment?

Reply:
Yes, EndOfSegmentException is the correct way of handling this. Updated the code.